File: src/Application Specific/Classes/Entity Collections/Order/Order.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace Orders

        Public Class Order

            Private _dataTypeValidator As DataTypeValidator

            Public Sub New()
                _dataTypeValidator = New DataTypeValidator
            End Sub

#Region "Class Properties"

            Public Property IgnoreExceptionsOnSetMethods() As Boolean
                Get
                    Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
                End Get
                Set(ByVal Value As Boolean)
                    Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
                End Set
            End Property

            Public ReadOnly Property Errors() As Hashtable
                Get
                    Return _dataTypeValidator.Errors
                End Get
            End Property

            Public ReadOnly Property DTValidator() As DataTypeValidator
                Get
                    Return _dataTypeValidator
                End Get
            End Property

            Private _exists As Boolean = False

            Public Property Exists() As Boolean
                Get
                    Return _exists
                End Get
                Set(ByVal Value As Boolean)
                    _exists = Value
                End Set
            End Property

            Private _deleted As Boolean = False

            Public ReadOnly Property Deleted() As Boolean
                Get
                    Return _deleted
                End Get
            End Property

            Public WriteOnly Property SetDeleted() As Boolean
                Set(ByVal Value As Boolean)
                    _deleted = Value
                End Set
            End Property

#End Region

#Region "Order Properties"

            Private _OrderID As Integer = 0

            Public ReadOnly Property OrderID() As Integer
                Get
                    Return _OrderID
                End Get
            End Property

            Public WriteOnly Property SetOrderID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_OrderID", Value)
                End Set
            End Property

            Private _DatePlaced As DateTime = Nothing

            Public Property DatePlaced() As DateTime
                Get
                    Return _DatePlaced
                End Get
                Set(ByVal Value As DateTime)
                    _DatePlaced = Value
                End Set
            End Property

            Private _OrderTypeID As Integer = 0

            Public ReadOnly Property OrderTypeID() As Integer
                Get
                    Return _OrderTypeID
                End Get
            End Property

            Public WriteOnly Property SetOrderTypeID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_OrderTypeID", Value)
                End Set
            End Property

            Private _OrderReference As String = String.Empty

            Public ReadOnly Property OrderReference() As String
                Get
                    Return _OrderReference
                End Get
            End Property

            Public WriteOnly Property SetOrderReference() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_OrderReference", Value)
                End Set
            End Property

            Private _UserID As Integer = 0

            Public ReadOnly Property UserID() As Integer
                Get
                    Return _UserID
                End Get
            End Property

            Public WriteOnly Property SetUserID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_UserID", Value)
                End Set
            End Property

            Private _OrderStatusID As Integer = 0

            Public ReadOnly Property OrderStatusID() As Integer
                Get
                    Return _OrderStatusID
                End Get
            End Property

            Public WriteOnly Property SetOrderStatusID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_OrderStatusID", Value)
                End Set
            End Property

            Private _ReasonForReject As String = String.Empty

            Public ReadOnly Property ReasonForReject() As String
                Get
                    Return _ReasonForReject
                End Get
            End Property

            Public WriteOnly Property SetReasonForReject() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_ReasonForReject", Value)
                End Set
            End Property

            Private _DeliveryDeadline As DateTime = Nothing

            Public Property DeliveryDeadline() As DateTime
                Get
                    Return _DeliveryDeadline
                End Get
                Set(ByVal Value As DateTime)
                    _DeliveryDeadline = Value
                End Set
            End Property

            Private _sentToSage As Boolean = False

            Public ReadOnly Property SentToSage() As Boolean
                Get
                    Return _sentToSage
                End Get
            End Property

            Public WriteOnly Property SetSentToSage() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_sentToSage", Value)
                End Set
            End Property

            Private _supplierInvoiceDate As DateTime = Nothing

            Public Property SupplierInvoiceDate() As DateTime
                Get
                    Return _supplierInvoiceDate
                End Get
                Set(ByVal Value As DateTime)
                    _supplierInvoiceDate = Value
                End Set
            End Property

            Private _supplierInvoiceAmount As Double = 0

            Public ReadOnly Property SupplierInvoiceAmount() As Double
                Get
                    Return _supplierInvoiceAmount
                End Get
            End Property

            Public WriteOnly Property SetSupplierInvoiceAmount() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_supplierInvoiceAmount", Value)
                End Set
            End Property

            Private _supplierInvoiceReference As String = ""

            Public ReadOnly Property SupplierInvoiceReference() As String
                Get
                    Return _supplierInvoiceReference
                End Get
            End Property

            Public WriteOnly Property SetSupplierInvoiceReference() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_supplierInvoiceReference", Value)
                End Set
            End Property

            Private _allocatedToUser As Integer = 0

            Public ReadOnly Property AllocatedToUser() As Integer
                Get
                    Return _allocatedToUser
                End Get
            End Property

            Public WriteOnly Property SetAllocatedToUser() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_allocatedToUser", Value)
                End Set
            End Property

            Private _invoiceAddressID As Integer = 0

            Public ReadOnly Property InvoiceAddressID() As Integer
                Get
                    Return _invoiceAddressID
                End Get
            End Property

            Public WriteOnly Property SetInvoiceAddressID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_invoiceAddressID", Value)
                End Set
            End Property

            Private _contactID As Integer = 0

            Public ReadOnly Property ContactID() As Integer
                Get
                    Return _contactID
                End Get
            End Property

            Public WriteOnly Property SetContactID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_contactID", Value)
                End Set
            End Property

            Private _specialInstructions As String = ""

            Public ReadOnly Property SpecialInstructions() As String
                Get
                    Return _specialInstructions
                End Get
            End Property

            Public WriteOnly Property SetSpecialInstructions() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_specialInstructions", Value)
                End Set
            End Property

            Private _Invoiced As Boolean = False

            Public ReadOnly Property Invoiced() As Boolean
                Get
                    Return _Invoiced
                End Get
            End Property

            Public WriteOnly Property SetInvoiced() As Boolean
                Set(ByVal Value As Boolean)
                    _Invoiced = Value
                End Set
            End Property

            Private _ExportedToSage As Boolean = False

            Public ReadOnly Property ExportedToSage() As Boolean
                Get
                    Return _ExportedToSage
                End Get
            End Property

            Public WriteOnly Property SetExportedToSage() As Boolean
                Set(ByVal Value As Boolean)
                    _ExportedToSage = Value
                End Set
            End Property

            Private _NominalCode As String = ""

            Public ReadOnly Property NominalCode() As String
                Get
                    Return _NominalCode
                End Get
            End Property

            Public WriteOnly Property SetNominalCode() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_NominalCode", Value)
                End Set
            End Property

            Private _DepartmentRef As String = ""

            Public ReadOnly Property DepartmentRef() As String
                Get
                    Return _DepartmentRef
                End Get
            End Property

            Public WriteOnly Property SetDepartmentRef() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_DepartmentRef", Value)
                End Set
            End Property

            Private _ExtraRef As String = ""

            Public ReadOnly Property ExtraRef() As String
                Get
                    Return _ExtraRef
                End Get
            End Property

            Public WriteOnly Property SetExtraRef() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_ExtraRef", Value)
                End Set
            End Property

            Private _TaxCodeID As Integer = 0

            Public ReadOnly Property TaxCodeID() As Integer
                Get
                    Return _TaxCodeID
                End Get
            End Property

            Public WriteOnly Property SetTaxCodeID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_TaxCodeID", Value)
                End Set
            End Property

            Private _ReadyToSendToSage As Boolean = False

            Public ReadOnly Property ReadyToSendToSage() As Boolean
                Get
                    Return _ReadyToSendToSage
                End Get
            End Property

            Public WriteOnly Property SetReadyToSendToSage() As Boolean
                Set(ByVal Value As Boolean)
                    _ReadyToSendToSage = Value
                End Set
            End Property

            Private _OrderConsolidationID As Integer = 0

            Public ReadOnly Property OrderConsolidationID() As Integer
                Get
                    Return _OrderConsolidationID
                End Get
            End Property

            Public WriteOnly Property SetOrderConsolidationID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_OrderConsolidationID", Value)
                End Set
            End Property

            Private _VATAmount As Double = 0.0

            Public ReadOnly Property VATAmount() As Double
                Get
                    Return _VATAmount
                End Get
            End Property

            Public WriteOnly Property SetVATAmount() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_VATAmount", Value)
                End Set
            End Property

            Private _DoNotConsolidated As Boolean = False

            Public ReadOnly Property DoNotConsolidated() As Boolean
                Get
                    Return _DoNotConsolidated
                End Get
            End Property

            Public WriteOnly Property SetDoNotConsolidated() As Object
                Set(ByVal value As Object)
                    _dataTypeValidator.SetValue(Me, "_DoNotConsolidated", value)
                End Set
            End Property

            Private _reason As String = String.Empty

            Public ReadOnly Property Reason() As String
                Get
                    Return _reason
                End Get
            End Property

            Public WriteOnly Property SetReason() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_reason", Value)
                End Set
            End Property

#End Region

        End Class

    End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Order/Order.cs
using System;
using System.Collections;

namespace FSM.Entity
{
    namespace Orders
    {
        public class Order
        {
            private DataTypeValidator _dataTypeValidator;

            public Order()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            
            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            private bool _exists = false;

            public bool Exists
            {
                get
                {
                    return _exists;
                }

                set
                {
                    _exists = value;
                }
            }

            private bool _deleted = false;

            public bool Deleted
            {
                get
                {
                    return _deleted;
                }
            }

            public bool SetDeleted
            {
                set
                {
                    _deleted = value;
                }
            }

            
            
            private int _OrderID = 0;

            public int OrderID
            {
                get
                {
                    return _OrderID;
                }
            }

            public object SetOrderID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_OrderID", value);
                }
            }

            private DateTime _DatePlaced = default;

            public DateTime DatePlaced
            {
                get
                {
                    return _DatePlaced;
                }

                set
                {
                    _DatePlaced = value;
                }
            }

            private int _OrderTypeID = 0;

            public int OrderTypeID
            {
                get
                {
                    return _OrderTypeID;
                }
            }

            public object SetOrderTypeID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_OrderTypeID", value);
                }
            }

            private string _OrderReference = string.Empty;

            public string OrderReference
            {
                get
                {
                    return _OrderReference;
                }
            }

            public object SetOrderReference
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_OrderReference", value);
                }
            }

            private int _UserID = 0;

            public int UserID
            {
                get
                {
                    return _UserID;
                }
            }

            public object SetUserID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_UserID", value);
                }
            }

            private int _OrderStatusID = 0;

            public int OrderStatusID
            {
                get
                {
                    return _OrderStatusID;
                }
            }

            public object SetOrderStatusID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_OrderStatusID", value);
                }
            }

            private string _ReasonForReject = string.Empty;

            public string ReasonForReject
            {
                get
                {
                    return _ReasonForReject;
                }
            }

            public object SetReasonForReject
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_ReasonForReject", value);
                }
            }

            private DateTime _DeliveryDeadline = default;

            public DateTime DeliveryDeadline
            {
                get
                {
                    return _DeliveryDeadline;
                }

                set
                {
                    _DeliveryDeadline = value;
                }
            }

            private bool _sentToSage = false;

            public bool SentToSage
            {
                get
                {
                    return _sentToSage;
                }
            }

            public object SetSentToSage
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_sentToSage", value);
                }
            }

            private DateTime _supplierInvoiceDate = default;

            public DateTime SupplierInvoiceDate
            {
                get
                {
                    return _supplierInvoiceDate;
                }

                set
                {
                    _supplierInvoiceDate = value;
                }
            }

            private double _supplierInvoiceAmount = 0;

            public double SupplierInvoiceAmount
            {
                get
                {
                    return _supplierInvoiceAmount;
                }
            }

            public object SetSupplierInvoiceAmount
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_supplierInvoiceAmount", value);
                }
            }

            private string _supplierInvoiceReference = "";

            public string SupplierInvoiceReference
            {
                get
                {
                    return _supplierInvoiceReference;
                }
            }

            public object SetSupplierInvoiceReference
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_supplierInvoiceReference", value);
                }
            }

            private int _allocatedToUser = 0;

            public int AllocatedToUser
            {
                get
                {
                    return _allocatedToUser;
                }
            }

            public object SetAllocatedToUser
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_allocatedToUser", value);
                }
            }

            private int _invoiceAddressID = 0;

            public int InvoiceAddressID
            {
                get
                {
                    return _invoiceAddressID;
                }
            }

            public object SetInvoiceAddressID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_invoiceAddressID", value);
                }
            }

            private int _contactID = 0;

            public int ContactID
            {
                get
                {
                    return _contactID;
                }
            }

            public object SetContactID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_contactID", value);
                }
            }

            private string _specialInstructions = "";

            public string SpecialInstructions
            {
                get
                {
                    return _specialInstructions;
                }
            }

            public object SetSpecialInstructions
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_specialInstructions", value);
                }
            }

            private bool _Invoiced = false;

            public bool Invoiced
            {
                get
                {
                    return _Invoiced;
                }
            }

            public bool SetInvoiced
            {
                set
                {
                    _Invoiced = value;
                }
            }

            private bool _ExportedToSage = false;

            public bool ExportedToSage
            {
                get
                {
                    return _ExportedToSage;
                }
            }

            public bool SetExportedToSage
            {
                set
                {
                    _ExportedToSage = value;
                }
            }

            private string _NominalCode = "";

            public string NominalCode
            {
                get
                {
                    return _NominalCode;
                }
            }

            public object SetNominalCode
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_NominalCode", value);
                }
            }

            private string _DepartmentRef = "";

            public string DepartmentRef
            {
                get
                {
                    return _DepartmentRef;
                }
            }

            public object SetDepartmentRef
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_DepartmentRef", value);
                }
            }

            private string _ExtraRef = "";

            public string ExtraRef
            {
                get
                {
                    return _ExtraRef;
                }
            }

            public object SetExtraRef
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_ExtraRef", value);
                }
            }

            private int _TaxCodeID = 0;

            public int TaxCodeID
            {
                get
                {
                    return _TaxCodeID;
                }
            }

            public object SetTaxCodeID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_TaxCodeID", value);
                }
            }

            private bool _ReadyToSendToSage = false;

            public bool ReadyToSendToSage
            {
                get
                {
                    return _ReadyToSendToSage;
                }
            }

            public bool SetReadyToSendToSage
            {
                set
                {
                    _ReadyToSendToSage = value;
                }
            }

            private int _OrderConsolidationID = 0;

            public int OrderConsolidationID
            {
                get
                {
                    return _OrderConsolidationID;
                }
            }

            public object SetOrderConsolidationID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_OrderConsolidationID", value);
                }
            }

            private double _VATAmount = 0.0;

            public double VATAmount
            {
                get
                {
                    return _VATAmount;
                }
            }

            public object SetVATAmount
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_VATAmount", value);
                }
            }

            private bool _DoNotConsolidated = false;

            public bool DoNotConsolidated
            {
                get
                {
                    return _DoNotConsolidated;
                }
            }

            public object SetDoNotConsolidated
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_DoNotConsolidated", value);
                }
            }

            private string _reason = string.Empty;

            public string Reason
            {
                get
                {
                    return _reason;
                }
            }

            public object SetReason
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_reason", value);
                }
            }

            
        }
    }
}